cfg/gke-1.6.0/node.yaml

```diff
@@ -0,0 +1,500 @@
+        audit: "/bin/ps -fC $kubeletbin"
+        audit_config: "/bin/cat $kubeletconf"
+        tests:
+          test_items:
+            - flag: --streaming-connection-idle-timeout
+              path: '{.streamingConnectionIdleTimeout}'
+              compare:
+                op: noteq
+                value: 0
+            - flag: --streaming-connection-idle-timeout
+              path: '{.streamingConnectionIdleTimeout}'
+              set: false
+          bin_op: or
+        remediation: |
+          Remediation Method 1:
+          If modifying the Kubelet config file, edit the kubelet-config.json file
+          /etc/kubernetes/kubelet-config.yaml and set the below parameter to a non-zero
+          value in the format of #h#m#s
+
+            "streamingConnectionIdleTimeout": "4h0m0s"
+
+          You should ensure that the kubelet service file
+          /etc/systemd/system/kubelet.service.d/10-kubelet-args.conf does not
+          specify a --streaming-connection-idle-timeout argument because it would
+          override the Kubelet config file.
+
+          Remediation Method 2:
+          If using executable arguments, edit the kubelet service file
+          /etc/systemd/system/kubelet.service.d/10-kubelet-args.conf on each
+          worker node and add the below parameter at the end of the KUBELET_ARGS variable
+          string.
+
+            --streaming-connection-idle-timeout=4h0m0s
+
+          Remediation Method 3:
+          If using the api configz endpoint consider searching for the status of
+          "streamingConnectionIdleTimeout": by extracting the live configuration from the
+          nodes running kubelet.
+          **See detailed step-by-step configmap procedures in Reconfigure a Node's Kubelet in a
+          Live Cluster (https://kubernetes.io/docs/tasks/administer-cluster/reconfigure-kubelet/),
+          and then rerun the curl statement from audit process to check for kubelet
+          configuration changes
+
+            kubectl proxy --port=8001 &
+            export HOSTNAME_PORT=localhost:8001 (example host and port number)
+            export NODE_NAME=gke-cluster-1-pool1-5e572947-r2hg (example node name from
+            "kubectl get nodes")
+            curl -sSL "http://${HOSTNAME_PORT}/api/v1/nodes/${NODE_NAME}/proxy/configz"
+
+          For all three remediations:
+          Based on your system, restart the kubelet service and check status
+
+            systemctl daemon-reload
+            systemctl restart kubelet.service
+            systemctl status kubelet -l
+        scored: true
+
+      - id: 3.2.6
+        text: "Ensure that the --make-iptables-util-chains argument is set to true (Automated)"
+        audit: "/bin/ps -fC $kubeletbin"
+        audit_config: "/bin/cat $kubeletconf"
+        tests:
+          test_items:
+            - flag: --make-iptables-util-chains
+              path: '{.makeIPTablesUtilChains}'
```

Comment:
@ttousai , here is the result, 
the iptables-util-chains is exists in the kubelet-config , but the test is failed and exoected result is empty, probably a typo in the test?



{
                            "test_number": "3.2.6",
                            "test_desc": "Ensure that the --make-iptables-util-chains argument is set to true (Automated)",
                            "audit": "/bin/ps -fC kubelet",
                            "AuditEnv": "",
                            "AuditConfig": "/bin/cat /etc/kubernetes/kubelet-config.yaml",
                            "type": "",
                            "remediation": "Remediation Method 1:\nIf modifying the Kubelet config file, edit the kubelet-config.json file\n/etc/kubernetes/kubelet/kubelet-config.json and set the below parameter to\ntrue\n\n  \"makeIPTablesUtilChains\": true\n\nEnsure that /etc/systemd/system/kubelet.service.d/10-kubelet-args.conf\ndoes not set the --make-iptables-util-chains argument because that would\noverride your Kubelet config file.\n\nRemediation Method 2:\nIf using executable arguments, edit the kubelet service file\n/etc/systemd/system/kubelet.service.d/10-kubelet-args.conf on each\nworker node and add the below parameter at the end of the KUBELET_ARGS variable\nstring.\n\n  --make-iptables-util-chains:true\n\nRemediation Method 3:\nIf using the api configz endpoint consider searching for the status of\n\"makeIPTablesUtilChains.: true by extracting the live configuration from the nodes\nrunning kubelet.\n\n**See detailed step-by-step configmap procedures in Reconfigure a Node's Kubelet in a\nLive Cluster (https://kubernetes.io/docs/tasks/administer-cluster/reconfigure-kubelet/),\nand then rerun the curl statement from audit process to check for kubelet\nconfiguration changes\n\n  kubectl proxy --port=8001 \u0026\n  export HOSTNAME_PORT=localhost:8001 (example host and port number)\n  export NODE_NAME=gke-cluster-1-pool1-5e572947-r2hg (example node name from\n  \"kubectl get nodes\")\n  curl -sSL \"http://${HOSTNAME_PORT}/api/v1/nodes/${NODE_NAME}/proxy/configz\"\n\nFor all three remediations:\nBased on your system, restart the kubelet service and check status\n\n  systemctl daemon-reload\n  systemctl restart kubelet.service\n  systemctl status kubelet -l\n",
                            "test_info": ["Remediation Method 1:\nIf modifying the Kubelet config file, edit the kubelet-config.json file\n/etc/kubernetes/kubelet/kubelet-config.json and set the below parameter to\ntrue\n\n  \"makeIPTablesUtilChains\": true\n\nEnsure that /etc/systemd/system/kubelet.service.d/10-kubelet-args.conf\ndoes not set the --make-iptables-util-chains argument because that would\noverride your Kubelet config file.\n\nRemediation Method 2:\nIf using executable arguments, edit the kubelet service file\n/etc/systemd/system/kubelet.service.d/10-kubelet-args.conf on each\nworker node and add the below parameter at the end of the KUBELET_ARGS variable\nstring.\n\n  --make-iptables-util-chains:true\n\nRemediation Method 3:\nIf using the api configz endpoint consider searching for the status of\n\"makeIPTablesUtilChains.: true by extracting the live configuration from the nodes\nrunning kubelet.\n\n**See detailed step-by-step configmap procedures in Reconfigure a Node's Kubelet in a\nLive Cluster (https://kubernetes.io/docs/tasks/administer-cluster/reconfigure-kubelet/),\nand then rerun the curl statement from audit process to check for kubelet\nconfiguration changes\n\n  kubectl proxy --port=8001 \u0026\n  export HOSTNAME_PORT=localhost:8001 (example host and port number)\n  export NODE_NAME=gke-cluster-1-pool1-5e572947-r2hg (example node name from\n  \"kubectl get nodes\")\n  curl -sSL \"http://${HOSTNAME_PORT}/api/v1/nodes/${NODE_NAME}/proxy/configz\"\n\nFor all three remediations:\nBased on your system, restart the kubelet service and check status\n\n  systemctl daemon-reload\n  systemctl restart kubelet.service\n  systemctl status kubelet -l\n"],
                            "status": "**FAIL**",
                            "actual_value": "apiVersion: kubelet.config.k8s.io/v1beta1\nkind: KubeletConfiguration\nauthentication:\n  webhook:\n    enabled: false\nauthorization:\n  mode: AlwaysAllow\nenableServer: false\nmakeIPTablesUtilChains:true\npodCIDR: 10.42.0.0/24\nstaticPodPath: /etc/kubernetes/manifests\nstaticPodURL: http://metadata.google.internal/computeMetadata/v1/instance/attributes/google-container-manifest\nstaticPodURLHeader:\n  Metadata-Flavor: [Google]\ncgroupDriver: systemd",
                            "scored": true,
                            "IsMultiple": false,
                            **"expected_result": ""**
                        }

Reply:
No it was a bad test, the test should PASS if the makeIPTablesUtilChains is not set on either the command line or in the config file which is our case so it should pass.

I have added a fix for the test.